bring/src/main/java/com/bobocode/petros/bring/factory/DefaultBeanFactory.java

```diff
@@ -0,0 +1,44 @@
+package com.bobocode.petros.bring.factory;
+
+import com.bobocode.petros.bring.context.domain.BeanDefinition;
+import com.bobocode.petros.bring.context.domain.BeanReference;
+import com.bobocode.petros.bring.context.domain.BeanScope;
+import com.bobocode.petros.bring.exception.NoSuchBeanDefinitionException;
+import com.bobocode.petros.bring.registry.BeanDefinitionRegistry;
+import lombok.SneakyThrows;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+public class DefaultBeanFactory implements BeanFactory {
+    private final BeanDefinitionRegistry beanDefinitionRegistry;
+
+    public DefaultBeanFactory(BeanDefinitionRegistry beanDefinitionRegistry) {
+        this.beanDefinitionRegistry = beanDefinitionRegistry;
+    }
+
+    @Override
+    public Map<String, BeanReference> getAllBeanReferences() {
+        return beanDefinitionRegistry.getAllBeanDefinitions().stream()
+                .collect(Collectors.toMap(BeanDefinition::getBeanName, this::createBeanReference));
+    }
+
+    @Override
+    @SneakyThrows
+    public BeanReference createBeanReference(final BeanDefinition beanDefinition) {
+        requireNonNull(beanDefinition);
+        if (beanDefinition.isInterface()) {
+            var beanDefinitionImpl = beanDefinition.getImplementations().values()
+                    .stream()
+                    .findFirst()
+                    .orElseThrow(() -> new NoSuchBeanDefinitionException("You must add an implementation of interface."));
```

Comment:
Please extract exception message to ExceptionMessages

Reply:
Basically, an exception is thrown, because an implementation is not found. Wouldn't it be better to have a message like this:
String.format("Implementation of '%s' is not found!", ((Class<?>) beanDefinition.getBeanClass()).getName()))